langchain4j-core/src/main/java/dev/langchain4j/exception/UnsupportedFeatureException.java

```diff
@@ -0,0 +1,11 @@
+package dev.langchain4j.exception;
+
+import dev.langchain4j.Experimental;
+
+@Experimental
+public class UnsupportedFeatureException extends RuntimeException {
```

Comment:
Do we need this? It seems just a duplicate of the Java's `UnsupportedOperationException`. Does it have any special purpose/semantic different from the native Java exception?

Reply:
The javadoc of the `UnsupportedOperationException` says:

>  * This class is a member of the
>  * a href="{@docRoot}/java.base/java/util/package-summary.html#CollectionsFramework"
>  * Java Collections Framework</a>.

Some argue that it should not be used outside of the collections. I have no hard opinion on this.

My main concern is different: the name of the exception says "this **_operation_** is not supported". There is only a single operation `ChatModel` can do: chat. So the **_operation_** is supported for sure. What is not supported hovewer is one of the input parameters that user specified (tools or json schema or some of the sampling parameters).

Maybe I am overthinking this and we can live with `UnsupportedOperationException`

Reply:
I'm not strongly opinionated on this, but even though it is true that it is originally part of the Collection Framework there are already tons of other libraries using that exception with a similar purpose and meaning. This suggestion only follows my main guideline when designing an API: `if in doubt leave it out` :smile:

Reply:
I repeat `if in doubt leave it out` to myself time and time again after watching your API design presentations :)

But does it really apply here? It is not about adding new functionality (which might be hard to support in the future), it is about using X or Y exception type. If X will prove wrong and I will want to use Y in the future, this will be a breaking change.

What are the downsides of using custom `UnsupportedFeatureException` instead of `UnsupportedOperationException` in your opinion?

Reply:
You're right, that applies to features and this is not a new feature. Nevertheless is code that can be avoided, other than that as I said I have no strong opinion on this.

Reply:
OK, thanks. I will keep `UnsupportedFeatureException` then as I think it is semantically more suitable than `UnsupportedOperationException` and I don't see any major drawbacks.